os.cpp

```diff
@@ -106,41 +106,41 @@ void scheduler()
 			if (!running || currentUser.id == sys) {
 				string input;
 				getline(cin, input);
-
 				switch (cmdToInt(input)) {
 					case 0: // "run"
-					if (currentUser.id != sys) {
-						machine.IR = main_memory[machine.PC];
-						// semwait();
-						interpreter();
-						// semsignal if necessary
-					} else {
-						cout << "Invalid command for system" << endl;
-					}
-					break;
+						if (currentUser.id != sys) {
+							machine.IR = main_memory[machine.PC];
+							// semwait();
+							interpreter();
+							// semsignal if necessary
+						} else {
+							cout << "Invalid command for system" << endl;
+						}
+						break;
 					case 1: // "dmp"
-					if (currentUser.id == sys) {
-						dump(false);
-					} else {
-						cout << "Invalid command for users" << endl;
-					}
-					break;
+						if (currentUser.id == sys) {
+							dump(false);
+						} else {
+							cout << "Invalid command for users" << endl;
+						}
+						break;
 					case 2: // "nop"
-					switchTime = sysclock;
-					break;
+						switchTime = sysclock;
+						break;
 					case 3: // "stp"
-					if (currentUser.id == sys) {
-						exit(EXIT_SUCCESS);
-					} else {
-						cout << "Invalid command for users" << endl;
-					}
-					break;
+						if (currentUser.id == sys) {
+							exit(EXIT_SUCCESS);
+						} else {
+							cout << "Invalid command for users" << endl;
+						}
+						break;
 					default:
-					cout << "Invalid command: " << input << endl;
-					break;
+						cout << "Invalid command: " << input << endl;
+						break;
 				}
 			}
 		}
+		dispatcher();
```

Comment:
I think this might need to come first. currentUser is not initialized anywhere, so `dispatcher()` must be called at least once before the scheduler starts trying to run anything.